torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,208 @@
+    major crops (maize, soybean, and rice)
+    in Northeast China from 2017 to 2019, using hierarchial mapping strategies,
+    random forest classifiers, interpolated and
+    smoothed 10-day Sentinel-2 time series data and
+    optimized features from spectral, temporal and
+    textural characteristics of the land surface.
+    The resultant maps have high overall accuracies (OA)
+    based on ground truth data. The dataset contains information
+    specific to three years: 2017, 2018, 2019.
+
+    The dataset contains 5 classes:
+
+    0. paddy rice
+    1. maize
+    2. soybean
+    3. others crops and lands
+    4. nodata
+
+    Dataset format:
+
+    * Three .TIF files containing the labels
+    * JavaScript code to download images from the dataset.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1038/s41597-021-00827-9
+
+    .. versionadded:: 0.6
+    """
+
+    filename_regex = r"CDL(?P<year>\d{4})_clip"
+    filename_glob = "CDL*.*"
+    zipfile_glob = "13090442.zip"
+
+    date_format = "%Y"
+    is_image = False
+    url = "https://figshare.com/ndownloader/articles/13090442/versions/1"
+    md5 = "eae952f1b346d7e649d027e8139a76f5"
+
+    cmap = {
+        0: (0, 255, 0, 255),
+        1: (255, 0, 0, 255),
+        2: (255, 255, 0, 255),
+        3: (128, 128, 128, 255),
+        15: (255, 255, 255, 255),
+    }
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        transforms: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new dataset.
+
+        Args:
+            paths: one or more root directories to search or files to load
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            FileNotFoundError: if no files are found in ``paths``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        self.paths = paths
+        self.download = download
+        self.checksum = checksum
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
```

Comment:
This is very minor, but I would actually use:
```suggestion
        self.ordinal_map = torch.full(max(self.cmap.keys()) + 1, 4, dtype=self.dtype)
```
The difference is that _if_ there happens to be a value we didn't expect, like 6, then it will be mapped to the background nodata class instead of paddy rice.

Reply:
Hi Adam, when I am trying to use the above line, it is throwing me this TypeError: '
 TypeError: full() received an invalid combination of arguments - got (int, int, dtype=torch.dtype), but expected one of:
 * (tuple of ints size, Number fill_value, *, tuple of names names, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)' . 
 How do I fix this?

Reply:
Just us a tuple instead of an int for size, so `(max(self.cmap.keys()),)`